protostar/commands/invoke_command.py

```diff
@@ -0,0 +1,145 @@
+from logging import Logger
+from typing import Any, Optional
+
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.signer import BaseSigner
+
+from protostar.cli import (
+    NetworkCommandUtil,
+    ProtostarArgument,
+    ProtostarCommand,
+    SignableCommandUtil,
+)
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet_gateway import GatewayFacadeFactory, SuccessfulInvokeResponse
+from protostar.starknet_gateway.gateway_facade import Fee
+
+
+class InvokeCommand(ProtostarCommand):
+    def __init__(
+        self,
+        logger: Logger,
+        gateway_facade_factory: GatewayFacadeFactory,
+    ):
+        self._logger = logger
+        self._gateway_facade_factory = gateway_facade_factory
+
+    @property
+    def name(self) -> str:
+        return "invoke"
+
+    @property
+    def description(self) -> str:
+        return "Sends an invoke transaction V1 to the StarkNet sequencer."
+
+    @property
+    def example(self) -> Optional[str]:
+        return None
+
+    @property
+    def arguments(self):
+        return [
+            *SignableCommandUtil.signable_arguments,
+            *NetworkCommandUtil.network_arguments,
+            ProtostarArgument(
+                name="contract-address",
+                description="The address of the contract being called.",
+                type="int",
+                is_required=True,
+            ),
+            ProtostarArgument(
+                name="function",
+                description="The name of the function being called.",
```

Comment:
```suggestion
                description="The name of the function from contract's interface.",
```

Reply:
https://docs.starknet.io/documentation/develop/CLI/commands/#starknet_invoke

Reply:
i understand, but if we can do better - we should :)

Reply:
@view defines contract interface yet (I think) can't be invoked